xapian-core/cluster/cluster.cc

```diff
@@ -246,12 +236,15 @@ Point::initialize(TermListGroup &tlg, const Document &doc_) {
     for (TermIterator it = doc.termlist_begin(); it != doc.termlist_end(); ++it) {
 	doccount wdf = it.get_wdf();
 	string term = *it;
-	if (wdf < 1 || term[0]=='Z')
+	double termfreq = tlg.get_termfreq(term);
+	/** If term is not a stemmed term or the term indexes only one
+	 *  one document, do not compute TF-IDF scores for them
+	 */
+	if (wdf < 1 || term[0] != 'Z' || termfreq <= 1)
```

Comment:
I'm happy to stick with hard-coded criteria for now. Using an `ExpandDecider` sounds like it would work, and I think would be a lot simpler than what I was proposing (the `wdf < 1` check would remain here, but that's to do with the compact vector calculation so it's fine).